src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -1112,6 +1112,235 @@ private static StringBuilder FormatStringBuilder(DateTime dateTime, ReadOnlySpan
+                        }
+
+                        return false;
+
+                    default:
+                        i++;
+                        break;
+                }
+            }
+
+            return true;
+        }
+
+
+        internal static bool IsValidCustomTimeFormat(ReadOnlySpan<char> format, bool throwOnError)
+        {
+            int length = format.Length;
+            int i = 0;
+
+            while (i < length)
+            {
+                switch (format[i])
+                {
+                    case '\\':
+                        if (i == length - 1)
+                        {
+                            if (throwOnError)
+                            {
+                                throw new FormatException(SR.Format_InvalidString);
+                            }
+
+                            return false;
+                        }
+
+                        i += 2;
+                        break;
+
+                    case '\'':
+                    case '"':
+                        char quoteChar = format[i++];
+                        while (i < length && format[i] != quoteChar)
+                        {
+                            i++;
+                        }
+
+                        if (i >= length)
+                        {
+                            if (throwOnError)
+                            {
+                                throw new FormatException(SR.Format(SR.Format_BadQuote, quoteChar));
+                            }
+
+                            return false;
+                        }
+
+                        i++;
+                        break;
+
+                    case 'd':
+                    case 'M':
+                    case 'y':
+                    case '/':
+                    case 'z':
+                    case 'k':
+                        if (throwOnError)
+                        {
+                            throw new FormatException(SR.Format_InvalidString);
+                        }
+
+                        return false;
+
+                    default:
+                        i++;
+                        break;
+                }
+            }
+
+            return true;
+        }
+
+        //   012345678901234567890123456789012
+        //   ---------------------------------
+        //   05:30:45.7680000
+        internal static bool TryFormatTimeOnlyO(int hour, int minute, int second, long fraction, Span<char> destination)
+        {
+            if (destination.Length < 16)
+            {
+                return false;
+            }
+
+            WriteTwoDecimalDigits((uint)hour, destination, 0);
+            destination[2] = ':';
+            WriteTwoDecimalDigits((uint)minute, destination, 3);
+            destination[5] = ':';
+            WriteTwoDecimalDigits((uint)second, destination, 6);
+            destination[8] = '.';
+            WriteDigits((uint)fraction, destination.Slice(9, 7));
```

Comment:
Nit: the `, 7` isn't necessary
```suggestion
            WriteDigits((uint)fraction, destination.Slice(9));
```

Reply:
will do.